tests/acceptance/features/bootstrap/DataExporterContext.php

```diff
@@ -256,14 +280,47 @@ private function assertFilePhysicallyExistInLastExport($filename, $message = '')
 	/**
 	 * @param string $filename
 	 * @param string $content
+	 * @param bool $trash
 	 *
 	 * @return void
 	 */
-	private function assertFilePhysicallyExistInLastExportWithContent($filename, $content) {
-		$this->featureContext->theFileWithContentShouldExistInTheServerRoot(
-			self::path("$this->lastExportPath/files/$filename"),
-			$content
-		);
+	private function assertFilePhysicallyExistInLastExportWithContent($filename, $content, $trash = false) {
+		if ($trash) {
+			$this->featureContext->listTrashbinFileInServerRoot(self::path("$this->lastExportPath/files_trashbin"));
+			$matches = [];
+			foreach (HttpRequestHelper::getResponseXml($this->featureContext->getResponse())->data->element as $item) {
+				$file = (string)$item;
+				if (\strpos($file, $filename) === 0) {
+					$matches[] = $file;
+				}
+			}
+			$fileFound = false;
+			foreach ($matches as $match) {
+				$path = "$this->lastExportPath/files_trashbin/$match";
+				$this->featureContext->readFileInServerRoot($path);
+				Assert::assertSame(
+					200,
+					$this->featureContext->getResponse()->getStatusCode(),
+					"Failed to read the file {$path}"
+				);
+				$fileContent = HttpRequestHelper::getResponseXml($this->featureContext->getResponse());
+				$fileContent = (string)$fileContent->data->element->contentUrlEncoded;
+				$fileContent = \urldecode($fileContent);
+
+				if ($fileContent === $content) {
+					$fileFound = true;
+					break;
+				}
+			}
+			if (!$fileFound) {
+				Assert::fail("error msg.");
```

Comment:
I ran a scenario to force the content being wrong:
```
  Scenario: Deleted file should be contained in an export                                                                         # /home/phil/git/owncloud/core/apps-external/data_exporter/tests/acceptance/features/cliDataExporter/export.feature:14
    Given user "user0" has created folder "testFolder"                                                                            # FeatureContext::userHasCreatedFolder()
    And user "user0" uploads file with content "content of trash bin file" to "/testFolder/trashbinFile.txt" using the WebDAV API # FeatureContext::userUploadsAFileWithContentTo()
    And user "user0" uploads file with content "hello" to "/testFolder/testfile.txt" using the WebDAV API                         # FeatureContext::userUploadsAFileWithContentTo()
    And user "user0" has deleted file "/testFolder/trashbinFile.txt"                                                              # FeatureContext::userHasDeletedFile()
    And as "user0" file "trashbinFile.txt" should exist in the trashbin                                                           # TrashbinContext::asFileOrFolderExistsInTrash()
    When user "user0" is exported to path "/tmp/fooSomething" using the occ command                                               # DataExporterContext::exportUserUsingTheOccCommand()
    Then the last export should contain file "/testFolder/testfile.txt" with content "hello"                                      # DataExporterContext::theLastExportShouldContainFileWithContent()
    And the last export should contain file "trashbinFile.txt" with content "file in trash zbin" in trashbin                      # DataExporterContext::theLastExportShouldContainFileWithContentInTrashbin()
      error msg.

--- Failed scenarios:

    /home/phil/git/owncloud/core/apps-external/data_exporter/tests/acceptance/features/cliDataExporter/export.feature:14

1 scenario (1 failed)
10 steps (9 passed, 1 failed)
```
The problem reported is just `error msg.` - that is not so helpful :)

If the filename is a problem then the code already gives me a message `File trashbinFile.txt not found in metadata`. So that is handled nicely already.

So the fail message here could report that the content is not what was expected. (And show the last example of content that did not match the expected string)

Reply:
e.g. you will have the last values in `$fileFoundWithMatchingContent` and `$content` so those could be reported in the fail message.

Reply:
fixed